rrmngmnt/nmcli.py

```diff
@@ -0,0 +1,644 @@
+# -*- coding: utf-8 -*-
+"""
+This module introduces support for the nmcli command which is provided by
+NetworkManager.
+"""
+import shlex
+
+from rrmngmnt.errors import CommandExecutionFailure
+from rrmngmnt.service import Service
+
+ERROR_MSG_FORMAT = "command -> {command}\nRC -> {rc}\nOUT -> {out}\nERROR -> {err}"  # noqa: E501
+
+NMCLI_COMMAND = "nmcli {options} {object} {command}"
+
+
+class NMCLI(Service):
+    """
+    This class implements network operations using nmcli.
+    """
+
+    def __init__(self, host):
+        super(NMCLI, self).__init__(host)
+        self._executor = host.executor()
+
+    def _exec_command(self, command):
+        """
+        Executes a command on the remote host.
+
+        Args:
+            command (str): a command to run remotely.
+
+        Returns:
+            str: command execution output.
+
+        Raises:
+            CommandExecutionFailure: if the remote host returned a code
+            indicating a failure in execution.
+        """
+        split = shlex.split(command)
+
+        rc, out, err = self._executor.run_cmd(split)
+
+        if rc != 0:
+            self.logger.error(
+                ERROR_MSG_FORMAT.format(command=command, rc=rc, out=out, err=err)  # noqa: E501
```

Comment:
either adjust the line max length or adjust your code to the project line length and remove  # noqa: E501